packages/autocomplete-core/src/onInput.ts

```diff
@@ -38,6 +43,10 @@ export function onInput<TItem extends BaseItem>({
   store,
   ...setters
 }: OnInputParams<TItem>): Promise<void> {
+  nextStateRef = nextState;
+  propsRef = props;
+  queryRef = query;
```

Comment:
All three values are used in the `then` callback. But because callbacks can be called in a different order than the call order, the values could be stale.